hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -7066,6 +7095,11 @@ else if ( getLoadQueryInfluencers().hasEnabledFetchProfiles() ) {
 							.getCollectionSemantics()
 							.getCollectionClassification();
 					if ( collectionClassification == CollectionClassification.BAG ) {
+						if ( containsCardinalityAlteringJoin && !(pluralAttributeMapping.getCollectionDescriptor() instanceof OneToManyPersister)) {
```

Comment:
We should only consider real bags in the `currentBagRole`

```suggestion
					if ( isRealBag( pluralAttributeMapping ) ) {
						if ( containsCardinalityAlteringJoin ) {
```